src/tiledb/cloud/file_ingestion/ingestion.py

```diff
@@ -0,0 +1,226 @@
+    """
+    logger = get_logger_wrapper(verbose)
+
+    # Flatten the results into a list.
+    flattened_results = [result for udf_result in udf_results for result in udf_result]
+
+    # Reduce batch size if there are fewer sample URIs
+    batch_size = min(batch_size, len(flattened_results))
+    num_chunks = ceil(len(flattened_results) / batch_size)
+    logger.info(f"Splitting results into {num_chunks} chunks...")
+
+    return [
+        flattened_results[n * batch_size : (n + 1) * batch_size]
+        for n in range(num_chunks)
+    ]
+
+
+def ingest_files_udf(
+    dataset_uri: str,
+    file_uris: Sequence[str],
+    destination: str,
+    *,
+    acn: Optional[str] = None,
+    config: Optional[dict] = None,
+    namespace: Optional[str] = None,
+    verbose: bool = False,
+) -> None:
+    """
+    Ingest files.
+
+    :param dataset_uri: The dataset URI.
+    :param file_uris: An iterable of file URIs.
+    :param destination: URI to ingest the files into.
+    :param acn: Access Credentials Name (ACN) registered in TileDB Cloud (ARN type),
+        defaults to None
+    :param config: Config dictionary, defaults to None.
+    :param namespace: TileDB-Cloud namespace, defaults to None.
+    :param verbose: Verbose logging, defaults to False.
+    """
+    logger = get_logger_wrapper(verbose)
+    # Trim potential trailing / in the destination URI
+    destination = destination.rstrip("/")
+
+    with tiledb.scope_ctx(config):
+        try:
+            if dataset_uri:
+                is_group = tiledb.object_type(dataset_uri) == "group"
+            else:
+                is_group = False
+        except Exception as exc:
+            # tiledb.object_type raises an exception
+            # if the namespace does not exist
+            logger.error(f"Error checking if {dataset_uri} is a Group. Bad namespace?")
+            raise exc
+
+        for file_uri in file_uris:
+            filename = sanitize_filename(os.path.basename(file_uri))
+            filestore_array_uri = f"{destination}/{filename}"
+            namespace = (
+                namespace or tiledb.cloud.user_profile().default_namespace_charged
+            )
+            logger.debug(
+                f"""
+                ---------------------------------------------
+                - Filename: {filename}
+                - Namespace: {namespace}
+                - Destination URI: {filestore_array_uri}
+                ---------------------------------------------"""
+            )
+
+            if not tiledb.array_exists(filestore_array_uri):
+                tiledb.cloud.file.create_file(
+                    namespace=namespace,
+                    name=filename,
+                    input_uri=file_uri,
+                    output_uri=filestore_array_uri,
+                    access_credentials_name=acn,
+                )
+
+                if is_group:
+                    with tiledb.Group(dataset_uri, mode="w") as group:
```

Comment:
Don't add to the group one at a time. Do it in a single batch at the end.

Reply:
After researching a bit this request, I concluded that I can do that by using the V2 API's `update_group_contents` from [here](https://github.com/TileDB-Inc/TileDB-Cloud-Py/blob/62524dd5c7cba945c754e4a7e7a65e7fa3302d05/src/tiledb/cloud/_common/api_v2/api/groups_api.py#L938).
Is this the correct way to approach that?